src/dataframe/dataframe.jl

```diff
@@ -337,6 +337,30 @@ function DataFrame(columns::AbstractMatrix, cnames::Symbol)
     return DataFrame(columns, gennames(size(columns, 2)), makeunique=false)
 end
 
+# Discontinued constructors
+
+DataFrame(matrix::Matrix) =
+    throw(ArgumentError("`DataFrame` constructor from a `Matrix` requires " *
+                        "passing :auto as a second argument to automatically " *
+                        "generate column names: `DataFrame(matrix, :auto)`"))
+
+DataFrame(matrix::Vector{<:AbstractVector}) =
+    throw(ArgumentError("`DataFrame` constructor from a `Vector` of vectors requires " *
+                        "passing :auto as a second argument to automatically " *
+                        "generate column names: `DataFrame(matrix, :auto)`"))
```

Comment:
```suggestion
DataFrame(vecs::AbstractVector{<:AbstractVector}) =
    throw(ArgumentError("`DataFrame` constructor from a `Vector` of vectors requires " *
                        "passing :auto as a second argument to automatically " *
                        "generate column names: `DataFrame(vecs, :auto)`"))
```

Reply:
Any idea why Codecov thinks this one isn't covered?

Reply:
I will check as I added tests for this. However, note that it must be a `Vector` not `AbstractVector` (as `AbstractVector`s are Tables.jl tables sometimes - just like above I use `Matrix` not `AbstractMatrix` for the same reason).

Reply:
this is tested